fuel-moshi/src/main/kotlin/com/github/kittinunf/fuel/moshi/FuelMoshi.kt

```diff
@@ -1,22 +1,25 @@
 package com.github.kittinunf.fuel.moshi
 
-import com.github.kittinunf.fuel.core.*
+import com.github.kittinunf.fuel.core.FuelError
+import com.github.kittinunf.fuel.core.Handler
+import com.github.kittinunf.fuel.core.Request
+import com.github.kittinunf.fuel.core.Response
+import com.github.kittinunf.fuel.core.ResponseDeserializable
+import com.github.kittinunf.fuel.core.response
 import com.github.kittinunf.result.Result
-import com.squareup.moshi.KotlinJsonAdapterFactory
 import com.squareup.moshi.Moshi
+import kotlin.reflect.KClass
 
 inline fun <reified T : Any> Request.responseObject(noinline handler: (Request, Response, Result<T, FuelError>) -> Unit) =
-        response(moshiDeserializerOf(), handler)
+        response(moshiDeserializerOf(T::class), handler)
 
-inline fun <reified T : Any> Request.responseObject(handler: Handler<T>) = response(moshiDeserializerOf(), handler)
+inline fun <reified T : Any> Request.responseObject(handler: Handler<T>) = response(moshiDeserializerOf(T::class), handler)
 
-inline fun <reified T : Any> Request.responseObject() = response(moshiDeserializerOf<T>())
+inline fun <reified T : Any> Request.responseObject() = response(moshiDeserializerOf(T::class))
 
-inline fun <reified T : Any> moshiDeserializerOf() = object : ResponseDeserializable<T> {
-    override fun deserialize(content: String): T? =
-            Moshi.Builder()
-                    .add(KotlinJsonAdapterFactory())
-                    .build()
-                    .adapter(T::class.java)
-                    .fromJson(content)
+fun <T : Any> moshiDeserializerOf(kClass: KClass<T>) = object : ResponseDeserializable<T> {
```

Comment:
[Q] What is the benefit of this version compared to the original version with usage of `reified` and `::class.java` ?

```
inline fun <reified T : Any> moshiDeserializerOf() = object : ResponseDeserializable<T> {
  override fun deserialize(content: String): T? = Moshi.Builder().build().adapter(T::class.java).fromJson(content) 
}
```
?

Reply:
@kittinunf, the original version is quite sucks when you looked at the bytecode level because you get FOUR Classes for this single files.
Mine version is quite Smaller than the original version.

Reply:
Ok then, should we also use the javaClass version so that we can avoid use the `kotlin.KClass` directly?

``` Kotlin
fun <T : Any> moshiDeserializerOf(clazz: Class<T>)
```

Reply:
Oh Okay I can do that.